libsolidity/formal/CHC.cpp

```diff
@@ -691,6 +699,57 @@ void CHC::visitAddMulMod(FunctionCall const& _funCall)
 	SMTEncoder::visitAddMulMod(_funCall);
 }
 
+void CHC::visitDeployment(FunctionCall const& _funCall)
+{
+	auto funType = dynamic_cast<FunctionType const*>(_funCall.expression().annotation().type);
+	ContractDefinition const* contract =
+		&dynamic_cast<ContractType const&>(*funType->returnParameterTypes().front()).contractDefinition();
+
+	// copy state variables from m_currentContract to state.storage.
+	state().writeStateVars(*m_currentContract, state().thisAddress());
+	errorFlag().increaseIndex();
+
+	auto originalTx = state().tx();
+	auto txOrigin = state().txMember("tx.origin");
+	state().newTx();
+	// set the transaction sender as this contract
+	m_context.addAssertion(state().txMember("msg.sender") == state().thisAddress());
+	// set the origin to be the current transaction origin
+	m_context.addAssertion(state().txMember("tx.origin") == txOrigin);
+
+	auto newAddr = state().newThisAddress();
+
+	if (auto constructor = contract->constructor())
+	{
+		auto const& args = _funCall.sortedArguments();
+		auto const& params = constructor->parameters();
+		solAssert(args.size() == params.size(), "");
+		for (size_t i = 0; i < args.size(); ++i)
+			m_context.addAssertion(expr(*args.at(i)) == m_context.variable(*params.at(i))->currentValue());
```

Comment:
Maybe can use `symbolicArguments` or something similar to get rid of this block, looks duplicated

Reply:
Doesn't work because this isn't the same as a FunctionCall to an actual function